### PR TITLE
Remove progressed work

### PR DESCRIPTION
**Pull request by acdlite**
The current implementation of resuming work is buggy. The underlying model is also flawed. Rather than attempt to fix a flawed model, we'll scrap the feature entirely and add it back later.

The triangle fuzz tester is now passing. By keeping this test green, we should have greater confidence that changes do not introduce regressions.

See https://github.com/facebook/react/issues/8830 for the larger plan.

### CODE REVIEW

File: src/renderers/noop/ReactNoopEntry.js [hunk 1]
```diff
@@ -430,10 +430,10 @@ var ReactNoop = {
       if (fiber.updateQueue) {
         logUpdateQueue(fiber.updateQueue, depth);
       }
-      const childInProgress = fiber.progressedChild;
+      const childInProgress = fiber.child;
       if (childInProgress && childInProgress !== fiber.child) {
```
Comment (sebmarkbage) [inline src/renderers/noop/ReactNoopEntry.js]: This branch is now always dead. Might be better to just comment it out so we know what it is supposed to do.

Comment (acdlite) [inline src/renderers/noop/ReactNoopEntry.js]: Fixed

File: src/renderers/shared/fiber/ReactFiberBeginWork.js [hunk 2]
```diff
@@ -410,50 +379,30 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
...
-        // different "hidden" settings.
-        let child = workInProgress.progressedChild;
-        while (child !== null) {
-          // To ensure that this subtree gets its priority reset, the children
-          // need to be reset.
-          child.pendingWorkPriority = OffscreenPriority;
-          child = child.sibling;
-        }
-        return null;
-      }
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/ReactFiberBeginWork.js]: I'm having a hard time convincing myself this isn't needed anymore.

File: src/renderers/shared/fiber/ReactFiberClassComponent.js [hunk 3]
```diff
@@ -580,7 +583,8 @@ module.exports = function(
       oldProps === newProps &&
       oldState === newState &&
       !hasContextChanged() &&
-      !(updateQueue !== null && updateQueue.hasForceUpdate)
+      !(workInProgress.updateQueue !== null &&
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/ReactFiberClassComponent.js]: So this is re-reading the updateQueue, is that the purpose? Why this change?

File: src/renderers/shared/fiber/__tests__/ReactIncrementalPerf-test.js [hunk 4]
```diff
@@ -280,7 +280,7 @@ describe('ReactDebugFiberPerf', () => {
     expect(getFlameChart()).toMatchSnapshot();
   });
 
-  it('measures deprioritized work', () => {
+  xit('measures deprioritized work', () => {
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/__tests__/ReactIncrementalPerf-test.js]: Can we re-enable this?

File: src/renderers/shared/fiber/ReactFiberBeginWork.js (hunk 2 above)
Comment (acdlite) [inline src/renderers/shared/fiber/ReactFiberBeginWork.js]: I think it's not needed because we handle this during priority reset, here: https://github.com/acdlite/react/blob/9713bd40a6e4eb406b6f66b8a55c4bbd662d5ae3/src/renderers/shared/fiber/ReactFiberScheduler.js#L558-L564

File: src/renderers/shared/fiber/ReactFiberClassComponent.js (hunk 3 above)
Comment (acdlite) [inline src/renderers/shared/fiber/ReactFiberClassComponent.js]: `beginUpdateQueue` sometimes clones `workInProgress.updateQueue`, so we need to point to the latest version.

File: src/renderers/shared/fiber/__tests__/ReactIncrementalSideEffects-test.js [hunk 5]
```diff
@@ -691,106 +691,109 @@ describe('ReactIncrementalSideEffects', () => {
...
-        this.state = {active: false};
-        barInstances.push(this);
-      }
-      activate() {
-        this.setState({active: true});
-      }
-      render() {
-        ops.push('Bar');
-        return <span prop={this.state.active ? 'X' : this.props.idx} />;
+  xit(
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/__tests__/ReactIncrementalSideEffects-test.js]: This one is similar. We might be able to just change the test a bit?

**Comment by acdlite** [comment]
@sebmarkbage Re-enabled those tests

File: src/renderers/shared/fiber/__tests__/ReactIncrementalPerf-test.js [hunk 6]
```diff
@@ -382,7 +382,7 @@ describe('ReactDebugFiberPerf', () => {
     expect(getFlameChart()).toMatchSnapshot();
   });
 
-  it('deduplicates lifecycle names during commit to reduce overhead', () => {
+  xit('deduplicates lifecycle names during commit to reduce overhead', () => {
```
Comment (gaearon) [inline src/renderers/shared/fiber/__tests__/ReactIncrementalPerf-test.js]: Why do we need to disable this? I don't think it tests specifically progressed work.
